llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1972,7 +1972,10 @@ static std::optional<Instruction *> instCombineSVESDIV(InstCombiner &IC,
   ConstantInt *SplatConstantInt = dyn_cast_or_null<ConstantInt>(SplatValue);
   if (!SplatConstantInt)
     return std::nullopt;
+
   APInt Divisor = SplatConstantInt->getValue();
+  if (Divisor.abs().getZExtValue() == 1)
+    return std::nullopt;
```

Comment:
For the `+1` case, would it be wrong to always return `Vec` here?

Reply:
Yeah that makes sense. I guess the predicate wouldn't matter in that case since it would be a no-op on the whole vector